funsor/terms.py

```diff
@@ -64,6 +64,12 @@ def reflect(cls, *args):
     Construct a funsor, populate ``._ast_values``, and cons hash.
     This is the only interpretation allowed to construct funsors.
     """
+    if len(args) > len(cls._ast_fields):
```

Comment:
Does this logic need to be dulicated in `memoize`?

Reply:
Yes, I think so, good catch